### ISSUE TITLE
Not OK: perl@6627 +SUIDMAIL on powerpc-machten 4.1.4    (UNINSTALLED) [PATCH]

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2367]
Migrated from [rt.perl.org#3695](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3695) (status was 'resolved')

Searchable as RT3695$

**Comment by p5pRT** [comment]
### From domo@computer.org



\[Let's hope these long lines get through\.  But whole thing also 
provided as an attachment in case they don't\.\]

\./perl op/regmesg\.t
1\.\.35
ok 1
ok 2
\# For /\(?\<= x\{10000\}\)/\, expected&#8203;:
\#  Lookbehind longer than 255 not implemented at \<HERE\< mark in regex m/\(?\<= x\{10000\}\) \<\<\<HERE\<\<\< /
\# Got&#8203;:
\#  Quantifier in \{\,\} bigger than 2046 at \<HERE\< mark in regex m/\(?\<=x\{ \<\<\<HERE\<\<\< 10000\}\)/
\#
not ok 3

Perl for MachTen is built with a regular expression repeat count limit
of 2047 so as to avoid it blowing its unextendable stack\, and the test
script correctly picks this up\, yet fails even so because the repeat
count of 10000 on the lookbehind is large enough to exceed the limit\,
if not the default\.  Knocking the count down by a factor of ten
fixes the problem&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- perl@6627/t/op/regmesg.t~   Tue Aug 15 11:16:54 2000
+++ perl@6627/t/op/regmesg.t    Tue Aug 15 13:37:42 2000
@@ -30,7 +30,7 @@

  '/(?<= .*)/' =>  'Variable length lookbehind not implemented at {#} mark in regex m/(?<= .*){#}/',

- '/(?<= x{10000})/' => 'Lookbehind longer than 255 not implemented at {#} mark in regex m/(?<= x{10000}){#}/',
+ '/(?<= x{1000})/' => 'Lookbehind longer than 255 not implemented at {#} mark in regex m/(?<= x{1000}){#}/',

  '/(?@)/' => 'Sequence (?@...) not implemented at {#} mark in regex m/(?@{#})/',


Ahem.  This is perl-current patched up to 6627, whatever the 
following might say.
```
</details>


<details><summary>Perl Info</summary>

``` ---
Flags:
    category=install
    severity=none

Site configuration information for perl v5.6.0:

Configured by domo at Tue Aug 15 11:32:20 WET DST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=machten, osvers=4.1.4, archname=powerpc-machten
    uname='machten ppp100 5 0.5 powerpc '
    config_args='-Doptimize=-g -de'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-g', gccversion=2.8.1, gccosandvers=machten4
    cppflags='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -DDEBUGGING -I/usr/local/include'
    ccflags ='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -DDEBUGGING -I/usr/local/include'
    stdchar='char', d_stdstdio=undef, usevfork=true
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='ld', ldflags =' -Xlstack=1048576 -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib
    libs=-lndbm -lgdbm -ldb -lm -lc -lutil
    libc=/usr/lib/libc.a, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_none.xs, dlext=none, d_dlsymun=undef, ccdlflags=''
    cccdlflags='', lddlflags=''

Locally applied patches:
    SUIDMAIL - fixes for suidperl security


@INC for perl v5.6.0:
    lib
    /usr/local/lib/perl5/5.6.0/powerpc-machten
    /usr/local/lib/perl5/5.6.0
    /usr/local/lib/perl5/site_perl/5.6.0/powerpc-machten
    /usr/local/lib/perl5/site_perl/5.6.0
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/domo
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH=/usr/lib
    LOGDIR (unset)
    PATH=/sbin:/usr/sbin:/home/domo/bin:/usr/local/bin:/usr/bin:/bin:/usr/X11/bin:/usr/libexec
    PERL_BADLANG (unset)
    SHELL=/bin/sh
```
</details>

**Comment by p5pRT** [comment]
### From domo@computer.org



\./perl op/regmesg\.t
1\.\.35
ok 1
ok 2
\# For /\(?\<= x\{10000\}\)/\, expected&#8203;:
\#  Lookbehind longer than 255 not implemented at \<HERE\< mark in regex m/\(?\<= x\{10000\}\) \<\<\<HERE\<\<\< /
\# Got&#8203;:
\#  Quantifier in \{\,\} bigger than 2046 at \<HERE\< mark in regex m/\(?\<=x\{ \<\<\<HERE\<\<\< 10000\}\)/
\#
not ok 3

Perl for MachTen is built with a regular expression repeat count limit
of 2047 so as to avoid it blowing its unextendable stack\, and the test
script correctly picks this up\, yet fails even so because the repeat
count of 10000 on the lookbehind is large enough to exceed the limit\,
if not the default\.  Knocking the count down by a factor of ten
fixes the problem&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- perl@6627/t/op/regmesg.t~   Tue Aug 15 11:16:54 2000
+++ perl@6627/t/op/regmesg.t    Tue Aug 15 13:37:42 2000
@@ -30,7 +30,7 @@

  '/(?<= .*)/' =>  'Variable length lookbehind not implemented at {#} mark in regex m/(?<= .*){#}/',

- '/(?<= x{10000})/' => 'Lookbehind longer than 255 not implemented at {#} mark in regex m/(?<= x{10000}){#}/',
+ '/(?<= x{1000})/' => 'Lookbehind longer than 255 not implemented at {#} mark in regex m/(?<= x{1000}){#}/',

  '/(?@)/' => 'Sequence (?@...) not implemented at {#} mark in regex m/(?@{#})/',
```
</details>


<details><summary>Perl Info</summary>

``` ---
Flags:
    category=install
    severity=none

Site configuration information for perl v5.6.0:

Configured by domo at Tue Aug 15 11:32:20 WET DST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=machten, osvers=4.1.4, archname=powerpc-machten
    uname='machten ppp100 5 0.5 powerpc '
    config_args='-Doptimize=-g -de'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-g', gccversion=2.8.1, gccosandvers=machten4
    cppflags='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -DDEBUGGING -I/usr/local/include'
    ccflags ='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -DDEBUGGING -I/usr/local/include'
    stdchar='char', d_stdstdio=undef, usevfork=true
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='ld', ldflags =' -Xlstack=1048576 -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib
    libs=-lndbm -lgdbm -ldb -lm -lc -lutil
    libc=/usr/lib/libc.a, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_none.xs, dlext=none, d_dlsymun=undef, ccdlflags=''
    cccdlflags='', lddlflags=''

Locally applied patches:
    SUIDMAIL - fixes for suidperl security


@INC for perl v5.6.0:
    lib
    /usr/local/lib/perl5/5.6.0/powerpc-machten
    /usr/local/lib/perl5/5.6.0
    /usr/local/lib/perl5/site_perl/5.6.0/powerpc-machten
    /usr/local/lib/perl5/site_perl/5.6.0
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/home/domo
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH=/usr/lib
    LOGDIR (unset)
    PATH=/sbin:/usr/sbin:/home/domo/bin:/usr/local/bin:/usr/bin:/bin:/usr/X11/bin:/usr/libexec
    PERL_BADLANG (unset)
    SHELL=/bin/sh
```
</details>

**Comment by p5pRT** [comment]
### From @jhi

> Perl for MachTen is built with a regular expression repeat count limit
> of 2047 so as to avoid it blowing its unextendable stack\, and the test
> script correctly picks this up\, yet fails even so because the repeat
> count of 10000 on the lookbehind is large enough to exceed the limit\,
> if not the default\.  Knocking the count down by a factor of ten
> fixes the problem&#8203;:

Applied\, thanks\.
